/home/konstantin.smirnov/miniconda3/envs/project_env/lib/python3.9/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:654: Checkpoint directory /home/konstantin.smirnov/associative-recurrent-memory-transformer/lstm_experimental/checkpoints/binary_copy exists and is not empty.
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]

  | Name       | Type             | Params | Mode
--------------------------------------------------------
0 | embedding  | Embedding        | 32     | train
1 | lstm1      | LSTM             | 70.7 K | train
2 | activation | ReLU             | 0      | train
3 | lstm2      | LSTM             | 132 K  | train
4 | lstm3      | LSTM             | 132 K  | train
5 | lstm4      | LSTM             | 132 K  | train
6 | fc         | Linear           | 516    | train
7 | loss_fn    | CrossEntropyLoss | 0      | train
--------------------------------------------------------
467 K     Trainable params
0         Non-trainable params
467 K     Total params
1.870     Total estimated model params size (MB)
8         Modules in train mode
0         Modules in eval mode
Epoch 8:  81%|█████████████████████████████████████████████████████████████████████████                 | 1015/1250 [00:21<00:04, 48.21it/s, v_num=yb5l]
/home/konstantin.smirnov/miniconda3/envs/project_env/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:424: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.
/home/konstantin.smirnov/miniconda3/envs/project_env/lib/python3.9/site-packages/pytorch_lightning/utilities/data.py:78: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 64. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.
                                                                                                                                                        
/home/konstantin.smirnov/miniconda3/envs/project_env/lib/python3.9/site-packages/pytorch_lightning/utilities/data.py:78: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 16. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.
Epoch 0, global step 1250: 'val_loss' reached 0.41203 (best 0.41203), saving model to '/home/konstantin.smirnov/associative-recurrent-memory-transformer/lstm_experimental/checkpoints/binary_copy/len400-best-checkpoint-val_loss=0.41-v1.ckpt' as top 1
Epoch 1, global step 2500: 'val_loss' reached 0.40273 (best 0.40273), saving model to '/home/konstantin.smirnov/associative-recurrent-memory-transformer/lstm_experimental/checkpoints/binary_copy/len400-best-checkpoint-val_loss=0.40.ckpt' as top 1
Epoch 2, global step 3750: 'val_loss' reached 0.40126 (best 0.40126), saving model to '/home/konstantin.smirnov/associative-recurrent-memory-transformer/lstm_experimental/checkpoints/binary_copy/len400-best-checkpoint-val_loss=0.40.ckpt' as top 1
Epoch 3, global step 5000: 'val_loss' reached 0.38792 (best 0.38792), saving model to '/home/konstantin.smirnov/associative-recurrent-memory-transformer/lstm_experimental/checkpoints/binary_copy/len400-best-checkpoint-val_loss=0.39.ckpt' as top 1
Epoch 4, global step 6250: 'val_loss' reached 0.37583 (best 0.37583), saving model to '/home/konstantin.smirnov/associative-recurrent-memory-transformer/lstm_experimental/checkpoints/binary_copy/len400-best-checkpoint-val_loss=0.38.ckpt' as top 1
Epoch 5, global step 7500: 'val_loss' was not in top 1
Epoch 6, global step 8750: 'val_loss' reached 0.36810 (best 0.36810), saving model to '/home/konstantin.smirnov/associative-recurrent-memory-transformer/lstm_experimental/checkpoints/binary_copy/len400-best-checkpoint-val_loss=0.37.ckpt' as top 1
Epoch 7, global step 10000: 'val_loss' reached 0.36632 (best 0.36632), saving model to '/home/konstantin.smirnov/associative-recurrent-memory-transformer/lstm_experimental/checkpoints/binary_copy/len400-best-checkpoint-val_loss=0.37.ckpt' as top 1

Detected KeyboardInterrupt, attempting graceful shutdown ...
